### PR TITLE
[Locale][API] Additional locale code validation

### PR DESCRIPTION
**Pull request by NoResponseMate**
| Q               | A                                                            |
|-----------------|--------------------------------------------------------------|
| Branch?         | 1.12 |
| Bug fix?        | yes                                                       |
| New feature?    | no                                                       |
| BC breaks?      | no                                                       |
| Deprecations?   | no |
| Related tickets | -                      |
| License         | MIT                                                          |

### CODE REVIEW

**Review by coldic3 — APPROVED** [review]

File: src/Sylius/Behat/Context/Api/Admin/ManagingLocalesContext.php [hunk 1]
```diff
@@ -44,6 +44,14 @@ public function iChoose(string $localeCode = ''): void
         $this->client->addRequestData('code', $localeCode);
     }
 
+    /**
+     * @When I set code to :code
+     */
```
Comment (jakubtobiasz) [inline src/Sylius/Behat/Context/Api/Admin/ManagingLocalesContext.php]: We can put this `@When` directly over `iChoose` method.

**Review by jakubtobiasz — APPROVED** [review]

File: src/Sylius/Behat/Context/Api/Admin/ManagingLocalesContext.php [hunk 2]
```diff
@@ -37,6 +37,7 @@ public function iWantToAddNewLocale(): void
 
     /**
      * @When I choose :localeCode
+     * @When I set code to :code
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Api/Admin/ManagingLocalesContext.php:40]:
```suggestion
     * @When I set code to :localeCode
```

File: src/Sylius/Bundle/LocaleBundle/Resources/translations/validators.en.yml [hunk 3]
```diff
@@ -3,4 +3,5 @@ sylius:
         code:
             not_blank: Please choose locale code.
             regex: Locale code can only be comprised of letters, numbers, dashes and underscores.
+            locale: This value is not a valid locale.
```
Comment (GSadee) [inline src/Sylius/Bundle/LocaleBundle/Resources/translations/validators.en.yml]:
```suggestion
            locale: This value is not a valid locale code.
```

File: src/Sylius/Behat/Context/Api/Admin/ManagingLocalesContext.php (hunk 2 above)
Comment (NoResponseMate) [reply src/Sylius/Behat/Context/Api/Admin/ManagingLocalesContext.php:40]: Won't work, the `Transform/LocaleContext` will grab it and throw a fatal.

File: features/locale/managing_locales/locale_validation.feature [hunk 4]
```diff
@@ -8,9 +8,23 @@ Feature: Locale validation
...
+
+    @api @no-ui
+    Scenario: Trying to add a locale with an invalid code
+        When I want to create a new locale
+        And I set code to "invalid"
+        And I try to add it
+        Then I should be notified that the code is invalid
+
+    @api @no-ui
+    Scenario: Trying to add a locale with an non-existent code
```
Comment (GSadee) [inline features/locale/managing_locales/locale_validation.feature:26]:
```suggestion
    Scenario: Trying to add a locale with a non-existent code
```

**Review by GSadee — APPROVED** [review]

**Comment by GSadee** [comment]
Thank you, Jan! :1st_place_medal:

**Pull request merged**